src/main/java/com/robinpowered/sdk/model/adapter/IdentifiableAdapter.java

```diff
@@ -0,0 +1,52 @@
+package com.robinpowered.sdk.model.adapter;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import com.robinpowered.sdk.model.IbeaconIdentifier;
+import com.robinpowered.sdk.model.Identifiable;
+import com.robinpowered.sdk.model.Identifier;
+
+import java.lang.reflect.Type;
+
+/**
+ * Handles adapting {@link com.robinpowered.sdk.model.Identifiable} from JSON representations.
+ */
+public class IdentifiableAdapter implements JsonDeserializer<Identifiable>,
+        JsonSerializer<Identifiable> {
+
+    @Override
+    public Identifiable deserialize(
+            JsonElement json,
+            Type typeOfT,
```

Comment:
Aren't we deserializing identifiers every time we get a JSON response from the API containing an identifier?

> What is this value equal to here

The parameters here are overridden - this is part of the super method but is unused by us. What do you mean?